packages/api-derive/src/currencies/balance.ts

```diff
@@ -0,0 +1,23 @@
+import { ApiInterfaceRx } from '@polkadot/api/types';
+
+import { AccountId, Balance } from '@orml/types/interfaces';
+import { Observable } from 'rxjs';
+import { memo } from '../util';
+
+export function balance(
+  api: ApiInterfaceRx
+): (address: AccountId | string | Uint8Array, token: any) => Observable<Balance> {
+  return memo(
+    (address: AccountId | string | Uint8Array, token: any): Observable<Balance> => {
+      const CurrencyId = api.registry.get('CurrencyId');
+      const currencyId = new (CurrencyId as any)(api.registry, token);
+      const nativeCurrencyId = api.consts.currencies.nativeCurrencyId;
+
+      if (currencyId.toHex() === nativeCurrencyId.toHex()) {
```

Comment:
```suggestion
      if (currencyId.eq(nativeCurrencyId)) {
```
I think there is `eq` does the right comparison